File: Python/factorial.py
def fact(num):
    if num < 0:
        return 0

    elif num == 0 or num == 1:
        return 1

    else:
        fact = 1
        while(num > 1):
            fact *= num       # Alternative: fact = fact * num
            num -= 1          # Alterantive: num = num - 1
        return fact

# User Input
num = int(input("Enter a number: "))
result = fact(num)
print("The factorial of",num,"is :",result)

File: c/factorial.c
// You can calculate the factorial of a number in C using a recursive or iterative approach. Here's an example of both methods:

//1. Using Recursion:
#include <stdio.h>

// Function to calculate the factorial of a number using recursion
int factorial(int n) {
    if (n == 0) {
        return 1; // 0! is defined as 1
    } else {
        return n * factorial(n - 1);
    }
}

int main() {
    int num;
    printf("Enter a non-negative integer: ");
    scanf("%d", &num);

    if (num < 0) {
        printf("Factorial is not defined for negative numbers.\n");
    } else {
        int result = factorial(num);
        printf("Factorial of %d is %d\n", num, result);
    }

    return 0;
}

//2. Using Iteration:

#include <stdio.h>

// Function to calculate the factorial of a number using iteration
int factorial(int n) {
    int result = 1;
    for (int i = 1; i <= n; i++) {
        result *= i;
    }
    return result;
}

int main() {
    int num;
    printf("Enter a non-negative integer: ");
    scanf("%d", &num);

    if (num < 0) {
        printf("Factorial is not defined for negative numbers.\n");
    } else {
        int result = factorial(num);
        printf("Factorial of %d is %d\n", num, result);
    }

    return 0;
}
